### PR TITLE
feat: useTokenList

### PR DESCRIPTION
**Pull request by angelia-yuqi-personal**
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/rigel/30)

### CODE REVIEW

File: src/utils/api.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import jsonRpc from './request'
+
+export function getAllTokenList(url) {
```
Comment (zctocm) [inline src/utils/api.js:3]: maybe we can rename all the API functions' name to a name with the same prefix, for example, requestTokenList, requestToken, then we know this is a function that fetches data from a remote backend

File: src/hooks/useTokenList.js [hunk 2]
```diff
@@ -1,5 +1,112 @@
...
+
+// get all token list from backend
+export function useAllTokenList() {
+  const {data} = useSWR(ProxyUrlPrefix.shuttleflow, getAllTokenList, {
+    refreshInterval: 60000,
+  })
+  return data
+}
+
+// chain token list
```
Comment (zctocm) [inline src/hooks/useTokenList.js:14]: the comment should be 'token list of one chain'

File: src/hooks/useTokenList.js [hunk 3]
```diff
@@ -1,5 +1,112 @@
...
+      toChain,
+      search,
+    ],
+    searchToken,
+  )
+  return data ? [data] : []
+}
+
+// search token adddress from current list
+function useSearchAddressFromsList(chain, search) {
```
Comment (zctocm) [inline src/hooks/useTokenList.js:40]: the `useSearchAddressFromsList` shoule be `useSearchAddressFromList `

File: src/hooks/useTokenList.js [hunk 4]
```diff
@@ -1,5 +1,112 @@
...
+  const searchAddressFromList = useSearchAddressFromsList(chain, lowerSearch)
+  const searchAddressFromBackend = useSearchAddressFromBackend(
+    chain,
+    lowerSearch,
+    fromChain,
+    toChain,
+  )
+  const searchNameFromList = useSearchNameFromList(chain, lowerSearch)
+
+  if (searchAddressFromList.length === 1) return searchAddressFromList
```
Comment (zctocm) [inline src/hooks/useTokenList.js:86]: I think the length should be >=1 because it is a Fuzzy search

File: src/hooks/useTokenList.js [hunk 5]
```diff
@@ -1,5 +1,112 @@
...
+  if (searchAddressFromBackend.length === 1) return searchAddressFromBackend
```
Comment (zctocm) [inline src/hooks/useTokenList.js:87]: same with the above

File: src/hooks/useTokenList.js [hunk 6]
```diff
@@ -1,5 +1,112 @@
...
+  if (searchNameFromList.length === 1) return searchNameFromList
+  return []
+}
+
+export function useCommonTokens(chain) {
+  const tokenList = useTokenList(chain)
+  const commonTokens = Config[chain].commonTokens
+  return tokenList.filter(obj => {
+    if (chain === ChainShortNameCfx) {
+      return commonTokens.filter(token => token === obj?.ctoken)
```
Comment (zctocm) [inline src/hooks/useTokenList.js:97]: should be `token?.toLowerCase() ===obj?.ctoken`

File: src/hooks/useTokenList.js [hunk 7]
```diff
@@ -1,5 +1,112 @@
...
+    } else {
+      return commonTokens.filter(token => token === obj?.reference)
```
Comment (zctocm) [inline src/hooks/useTokenList.js:99]: same with above

File: src/hooks/useTokenList.js (hunk 4 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js:86]: search address  is not the fuzzy search, but search name is fuzzy.

File: src/utils/api.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  return jsonRpc(url, 'getTokenList', {})
+}
+
+export function searchToken(url, fromChain, toChian, address) {
```
[the remark this answers is not in the document]
Comment (angelia-yuqi-personal) [reply src/utils/api.js]: done

File: src/utils/api.js (hunk 1 above)
Comment (angelia-yuqi-personal) [reply src/utils/api.js]: done

File: src/hooks/useTokenList.js (hunk 2 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js]: done

File: src/hooks/useTokenList.js (hunk 3 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js]: done

File: src/hooks/useTokenList.js (hunk 4 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js:86]: done for search name

File: src/hooks/useTokenList.js (hunk 6 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js:97]: token is the backend data. and it's always lowercase

File: src/hooks/useTokenList.js (hunk 7 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js:99]: same with above

File: src/hooks/useTokenList.js (hunk 6 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js:97]: image

File: src/hooks/useTokenList.js (hunk 4 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js:86]: `searchAddressFromList` should return only one item array

File: src/hooks/useTokenList.js (hunk 5 above)
Comment (angelia-yuqi-personal) [reply src/hooks/useTokenList.js:87]: same with the above

**Review by zctocm — DISMISSED** [review]

**Review by zctocm — APPROVED** [review]

**Pull request merged**